### PR TITLE
eukleides: specify texlive dependencies instead of buildInputs

### PR DESCRIPTION
**Pull request by xworld21**
###### Motivation for this change
The use of `texLive` as build input was suspicious, and it turns out it is not right. If the user is calling e.g. `euktopdf`, it is always in the context of the user compiling a tex file, so `euktopdf` should call the user's version, not its own copy. Also the default value of `texLive` did not include the necessary dependencies, so it would not work anyway.

~`eukleides` is still partly broken, for instance `epstopdf` must be called with an extra option after recent changes in TeX Live. It would be a good idea to report this upstream.~

Edit: the PR includes a few bug fixes, one of which to be reported upstream.

@SFrijters I am submitting this as a draft, do you think you can improve it, and maybe do some testing? I don't use this myself. You may have a better idea of how the package should be used.

Cc @peti 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SFrijters** [comment]
@xworld21 Just to clarify, I just picked this package for ZHF and because I use TeX from time to time it seemed like something useful to know more about, but I've never used this package myself (and I hadn't even heard of it before last week), so I don't have an actual use case. Nevertheless, I can take a look later, but it would also just be try-as-I-go.

**Comment by xworld21** [comment]
@peti as maintainer, do you know why `texLive` was a build input? As far as I can tell, the package works without it, but maybe I missed some particular use case.

I have also fixed a few other problems. I will report the `-dALLOWPSTRANSPARENCY` issue upstream.

**Comment by peti** [comment]
> @peti as maintainer, do you know why `texLive` was a build input?

Sorry, but I don't know. It was already listed in the build inputs when I took an interest in the package. I assumed it was required for building the package's documentation, but I just guessing really.

**Review by peti — APPROVED** [review]
Looks good to me

**Comment by SFrijters** [comment]
@xworld21 Haven't had time to look at this, and it would probably be next weekend at the earliest, so don't wait for me to check it if it works for you.

**Comment by xworld21** [comment]
> I assumed it was required for building the package's documentation, but I just guessing really.

That could be the case, not every package brings its own texlive, maybe `texinfo` was like this.

At any rate, the core binaries (`eukleides` and `euktopst`) work without texlive. `euktoeps` could benefit from wrapping `scheme-small`, but it's another place where the user's texlive might be a better choice.

I have sent an email to the creator about the transparency fix.

**Comment by xworld21** [comment]
I believe this PR is as good as it gets, I won't be making further changes. It works in the (very few) tests I have run.

File: pkgs/applications/science/math/eukleides/default.nix [hunk 1]
```diff
@@ -34,19 +38,16 @@ lib.fix (eukleides: stdenv.mkDerivation rec {
...
   postInstall = ''
     wrapProgram $out/bin/euktoeps \
-      --set-default TEXINPUTS : \
-      --prefix TEXINPUTS : "$tex/tex/latex/eukleides" \
-      --prefix PATH : "${texLive}/bin"
-    wrapProgram $out/bin/euktopdf \
-      --set-default TEXINPUTS : \
-      --prefix TEXINPUTS : "$tex/tex/latex/eukleides" \
-      --prefix PATH : "${texLive}/bin"
+      --prefix PATH : ${getopt}/bin
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/eukleides/default.nix:41]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ getopt ]}
```

File: pkgs/applications/science/math/eukleides/default.nix [hunk 2]
```diff
@@ -34,19 +38,16 @@ lib.fix (eukleides: stdenv.mkDerivation rec {
...
   '';
 
   outputs = [ "out" "doc" "tex" ];
 
   passthru.tlType = "run";
-  passthru.pkgs = [ eukleides.tex ];
+  passthru.pkgs = [ eukleides.tex ] ++
+    # packages needed by euktoeps, euktopdf and eukleides.sty
+    (with texlive; collection-pstricks.pkgs ++ epstopdf.pkgs
+      ++ iftex.pkgs ++ moreverb.pkgs);
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/eukleides/default.nix:50]:
```suggestion
  passthru.pkgs = [ eukleides.tex ]
    # packages needed by euktoeps, euktopdf and eukleides.sty
    ++ (with texlive; collection-pstricks.pkgs ++ epstopdf.pkgs ++ iftex.pkgs ++ moreverb.pkgs);
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/eukleides/default.nix (hunk 2 above)
Comment (xworld21) [reply pkgs/applications/science/math/eukleides/default.nix]: Thanks! Is this coming from `nixpkgs-fmt` or something similar? Happy to use it in the future, if that's the advice.

File: pkgs/applications/science/math/eukleides/default.nix (hunk 1 above)
Comment (xworld21) [reply pkgs/applications/science/math/eukleides/default.nix]: Squashed in the corresponding commit.

File: pkgs/applications/science/math/eukleides/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/math/eukleides/default.nix]: No, I just shortened it a bit.

**Pull request merged**